src/schema/v2/TargetSupply/index.ts

```diff
@@ -0,0 +1,67 @@
+import { ResolverContext } from "types/graphql"
+import { GraphQLObjectType, GraphQLFieldConfig, GraphQLList } from "graphql"
+import { ArtistType } from "../artist"
+import {
+  getMicrofunnelData,
+  getTargetSupplyArtists,
+} from "../artist/targetSupply/utils/getMicrofunnelData"
+import {
+  getRecentlySoldArtworksConnection,
+  RecentlySoldArtworksConnectionSource,
+} from "../types/recentlySoldArtworksConnection"
+import { TargetSupplyMicrofunnelMetadata } from "../types/targetSupplyMicrofunnelMetadata"
+
+interface TargetSupplyMicrofunnelItemSource
+  extends RecentlySoldArtworksConnectionSource {
+  slug: string
+}
+
+const TargetSupplyType = new GraphQLObjectType<any, ResolverContext>({
+  name: "TargetSupply",
+  fields: {
+    microfunnel: {
+      type: new GraphQLList(
+        new GraphQLObjectType<
+          TargetSupplyMicrofunnelItemSource,
+          ResolverContext
+        >({
+          name: "TargetSupplyMicrofunnelItem",
+          fields: {
+            metadata: {
+              type: TargetSupplyMicrofunnelMetadata,
+            },
+
+            artist: {
+              type: ArtistType,
+              resolve: async ({ slug }, {}, { artistLoader }) => {
+                const artist = await artistLoader(slug)
+                return artist
+              },
+            },
+
+            artworksConnection: getRecentlySoldArtworksConnection(),
```

Comment:
👍 nice

Reply:
To avoid the typescript error 
`Property 'type' is missing in type '() => GraphQLFieldConfig<RecentlySoldArtworksConnectionSource, ResolverContext, { [argName: string]: any; }>' but required in type 'GraphQLFieldConfig<TargetSupplyMicrofunnelItemSource, ResolverContext, { [key: string]: any; }>'.ts(2741)` 

and the runtime error 

`TargetSupplyMicrofunnelItem.artworksConnection field config must be an object`. 

I thought it would be fine without a getter function too, but it doesn't seem to be.

Reply:
No that totally makes sense, kinda similar to the `fields: () =>` thing we ran into earlier.